libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/PredictiveCacheController.kt

```diff
@@ -50,69 +53,80 @@ private const val RASTER_SOURCE_TYPE = "raster"
  * - `OnboardRouterOptions` enabled you to specify a path where nav-tiles will be saved and if a
  * custom directory was used, it should be cleared as well.
  *
- * @param navigation [MapboxNavigation] instance
+ * @param navigationPredictiveCacheLocationOptions [PredictiveCacheLocationOptions] location configuration for navigation predictive caching
  * @param predictiveCacheControllerErrorHandler [PredictiveCacheControllerErrorHandler] listener (optional)
  */
 class PredictiveCacheController @JvmOverloads constructor(
-    private val navigation: MapboxNavigation,
+    navigationPredictiveCacheLocationOptions: PredictiveCacheLocationOptions =
+        PredictiveCacheLocationOptions.Builder().build(),
     private val predictiveCacheControllerErrorHandler: PredictiveCacheControllerErrorHandler? = null
 ) {
-    private var map: MapboxMap? = null
-
-    private val onStyleLoadedListener = object : OnStyleLoadedListener {
-        override fun onStyleLoaded() {
-            map?.let { map ->
-                val tileStore = map.getResourceOptions().tileStore
-                if (tileStore == null) {
-                    handleError("TileStore instance not configured for the Map.")
-                    return
-                }
-                val currentMapSources = mutableListOf<String>()
-                traverseMapSources(map) { tileVariant ->
-                    currentMapSources.add(tileVariant)
-                }
-                updateMapsControllers(
-                    currentMapSources,
-                    PredictiveCache.currentMapsPredictiveCacheControllers(),
-                    tileStore
-                )
-            }
-        }
-    }
+    private var mapListeners = mutableMapOf<MapboxMap, OnStyleLoadedListener>()
 
     init {
-        PredictiveCache.createNavigationController(
-            navigation.navigationOptions.predictiveCacheLocationOptions
-        )
+        PredictiveCache.createNavigationController(navigationPredictiveCacheLocationOptions)
     }
 
     /**
-     * Call when a new map instance is available. Only one map instance at a time is supported.
+     * Create cache controllers for a map instance.
+     * Call when a new map instance is available.
+     *
+     * @param map an instance of [MapboxMap]
+     * @param cacheLocationOptions options to configure cache controllers
+     * @param sourceIdsToCache a list of sources to cache.
+     * Source id's should look like "mapbox://mapbox.satellite", "mapbox://mapbox.mapbox-terrain-v2".
+     * The system only supports source hosted on Mapbox Services which URL starts with "mapbox://".
+     * If no ids are passed all available style sources will be cached.
      */
-    fun setMapInstance(map: MapboxMap) {
-        removeMapInstance()
+    @JvmOverloads
+    fun createMapControllers(
+        map: MapboxMap,
+        cacheLocationOptions: PredictiveCacheLocationOptions =
+            PredictiveCacheLocationOptions.Builder().build(),
```

Comment:
This argument now becomes a little bit confusing when we have `PredictiveCacheLocationOptions` also passed in the constructor. Why would we ever want to cache more or less map tiles than routing tiles for example?

What if we use `PredictiveCacheLocationOptions` that are passed in the constructor for now and reconsider exposing more granularity when there's a clear use-case for it?

Reply:
ok, let's remove, we can add later if necessary